ydb/core/persqueue/pq_impl.cpp

```diff
@@ -2981,11 +3194,72 @@ bool TPersQueue::CanProcessDeleteTxs() const
     return !DeleteTxs.empty();
 }
 
+void TPersQueue::SubscribeWriteId(ui64 writeId,
+                                  const TActorContext& ctx)
+{
+    ctx.Send(NLongTxService::MakeLongTxServiceID(ctx.SelfID.NodeId()),
+             new NLongTxService::TEvLongTxService::TEvSubscribeLock(writeId, ctx.SelfID.NodeId()));
+}
+
+void TPersQueue::ForwardGetOwnershipToShadowPartitions(const TActorContext& ctx)
+{
+    for (auto& params : HandleGetOwnershipRequestParams) {
+        ui64 writeId = params.Request.GetWriteId();
+        ui32 partitionId = params.Request.GetPartition();
+        Y_ABORT_UNLESS(TxWrites.contains(writeId) && TxWrites[writeId].Partitions.contains(partitionId));
+        TTxWriteInfo& txWrite = TxWrites.at(writeId);
+        ui32 shadowPartitionId = txWrite.Partitions[partitionId];
+        Y_ABORT_UNLESS(ShadowPartitions.contains(shadowPartitionId));
+
+        CreateShadowPartitionActor(shadowPartitionId, ctx);
+
+        TPartitionInfo& partition = ShadowPartitions.at(shadowPartitionId);
+        partition.GetOwnershipRequests.emplace_back(params.Cookie, params.Request, params.Sender);
+
+        if (txWrite.LongTxSubscriptionStatus == NKikimrLongTxService::TEvLockStatus::STATUS_UNSPECIFIED) {
+            SubscribeWriteId(writeId, ctx);
```

Comment:
Subscribe to write id when got new writeid, not when in persisted

Reply:
done
https://github.com/ydb-platform/ydb/pull/1037/commits/74c832f44de40a1c11abdeb95e1b95dd7266a9bc